typing/typedtree.mli

```diff
@@ -250,12 +250,13 @@ and expression_desc =
       expression * Longident.t loc * Types.label_description * expression
   | Texp_array of expression list
   | Texp_ifthenelse of expression * expression * expression option
-  | Texp_sequence of expression * expression
+  | Texp_sequence of expression * Types.layout * expression
   | Texp_while of {
       wh_cond : expression;
       wh_cond_region : bool; (* False means allocates in outer region *)
       wh_body : expression;
-      wh_body_region : bool  (* False means allocates in outer region *)
+      wh_body_region : bool;  (* False means allocates in outer region *)
+      wh_body_layout : Types.layout
```

Comment:
This is very similar to the layout in `Texp_sequence`. I remain curious why it's necessary. (I'll stop writing this on other similar fields.)